### PR TITLE
[FIX] Cast X to inner mtype in custom public methods

### LINKED ISSUE #180: [BUG] X not casted to inner mtype in some public methods
I noticed an issue where the exogenous data `X` wasn't properly cast to the model's specified scitype when using `predict_component_samples`. I think a `X_inner = self._check_X(X=X)` should be in place for `_get_predictive_samples_dict`, no? I can create a PR if it's of interest.

### PR DESCRIPTION
**Pull request by tingiskhan**
Resolves #180

### CODE REVIEW

**Comment by felipeangelimvieira** [comment]
Thank you! I saw your comment

```python
# NB: the new inference engine behaviour causes issues with convergence, so we fix by seeding for now, but
# we might need to override r_hat in `simulate` if the inference engine has that property for a more sustainable fix
np.random.seed(123)
```

But I wasn't able to reproduce the errors with the code as it was below (without the seed). What errors were you observing exactly?

**Comment by felipeangelimvieira** [comment]
Oh I see - the default r_hat is breaking some usecases that were passing previously, isn't it? I think setting r_hat to jnp.inf will be a good practice - then it doesn't break already existent code and still makes it possible to check convergence. I'll do it in a separate PR.

**Comment by tingiskhan** [comment]
> Oh I see - the default r_hat is breaking some usecases that were passing previously, isn't it? I think setting default r_hat to jnp.inf is a good choice - then it doesn't break already existent code and still makes it possible to check convergence. I'll do it in a separate PR.

Yes, exactly! And I agree, I think setting it to none will skip check entirely
